rpcs3/rpcs3qt/pad_settings_dialog.cpp

```diff
@@ -117,6 +118,12 @@ pad_settings_dialog::pad_settings_dialog(QWidget *parent, const GameInfo *game)
 			LOG_ERROR(GENERAL, "Failed to convert device string: %s", m_device_name);
 			return;
 		}
+		// Update battery indicator (if one is present) if device selection is changed
+		if (m_enable_battery)
+		{
+			battery_level = m_handler->get_battery_level(m_device_name);
+			ui->pb_battery->setValue(battery_level);
```

Comment:
you should most likely move this section into fail_callback with
```
if (m_enable_battery)
{
    ui->pb_battery->setValue(0);
}
```

Reply:
I experimented with this, but it resets the progressbar even for connected controllers. I added this anyway, but I did not remove the referenced snippet. The current implementation works correctly and consistently with multiple controllers, and I haven't been successful at finding a different way to do this.